File: buffer/src/main/java/org/epdb/buffer/BufferManager.java
package org.epdb.buffer;

import org.epdb.buffer.dto.BufferFrame;
import org.epdb.storage.dto.Page;

public interface BufferManager {
    
    Page getPage(Long pageId);

    void unpinPage(Long pageId, boolean isModified);

    void flushPage(BufferFrame bufferFrame);

    Page allocateNewPage(int tableId);
}

File: buffer/src/main/kotlin/org/epdb/buffer/BufferManager.kt
package org.epdb.buffer

import org.epdb.buffer.dto.BufferFrame
import org.epdb.storage.dto.Page

interface BufferManager {
    fun getPage(pageId: Long): Page

    fun unpinPage(pageId: Long, isModified: Boolean)

    fun flushPage(bufferFrame: BufferFrame)

    fun allocateNewPage(tableId: Int): Page
}
